Eksctl usage documents (#12)
ajktown/docs: eksctl/01_setup.md

# eksctl setup

<!-- TOC -->

- [eksctl setup](#eksctl-setup)
  - [Overview](#overview)
  - [Tips](#tips)
  - [Install eksctl](#install-eksctl)
  - [Check your default region of AWS](#check-your-default-region-of-aws)
  - [Requirements for instance](#requirements-for-instance)
  - [Create cluster with node type](#create-cluster-with-node-type)
    - [Create Cluster (ARM)](#create-cluster-arm)
    - [Create Cluster  (x86)](#create-cluster--x86)
  - [Get Cluster](#get-cluster)
- [Delete Cluster](#delete-cluster)

<!-- /TOC -->

## Overview

> eksctl is a simple CLI tool for creating clusters on EKS - Amazon's new managed Kubernetes service for EC2. It is written in Go, and uses CloudFormation.

Author: ```suggestion
eksctl is a simple CLI tool for creating clusters on EKS. It will create separate vpc, security groups etc for your sake. 
```

eksctl is explained in a straightforward manner in [this video](https://youtu.be/p6xDCz00TxU).

> 

Author: ```suggestion
Undertaking these steps is estimated to cost around $70-$90 per month. However, this is an estimate and not a guaranteed figure. It is your responsible to manage the AWS cost. Instead of deleting every resource on AWS Console by your own, you can simply delete the cluster with eksctl command, and it will handle for you. You must double check if there any left overs that can cost you.
```

## Tips 

> If something does not go well as it is supposed to, I highly recommend to [delete cluster](#delete-cluster) and [create it again](#create-cluster-with-node-type).

Author: ```suggestion
If something does not go well as it is supposed to, I highly recommend to [delete cluster](#delete-cluster) and [create it again](#create-cluster-with-node-type). And then start from the beginning.
```

## Install eksctl

## Check your default region of AWS

eksctl uses your aws configured region on ~/.aws/config

``` sh
$ cat ~/.aws/config
# Output:
# [default]
# region = ap-northeast-1
```

If you do not change your region, you have to specify region for every command.

``` sh
eksctl get clusters --region ap-northeast-1
```

> We highly recommend to modify the region using `vi`

Author: ```suggestion
We highly recommend to modify the region using `vi`, to skip specifying your deploying region for every command.
```

> # eksctl setup

Author: TODOs
- [ ] Apply every change of the FE and API
- [ ] Write how to deploy ajktown application step by step (of what you did)

Author: handled.
